### PR TITLE
feat(rest) : Remove mail-request parameter and read from config file while downloading reports.

### LINKED ISSUE #2256: feat(Rest) : Remove mail-request parameter and read from config file while downloading reports.
[//]: # (Copyright Siemens AG, 2021. Part of the SW360 Portal Project)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

### Description

Remove mail-request parameter from rest end point and read the value(mailrequest is true or false based on UI configuration) 
from config file while downloading following reports.

1. Project report (with/without linkedReleases)
2. Component report  (with/without linkedReleases)
3. License report
4. License clearing report

#### How to reproduce

Describe the bug and list the steps you used when the issue occurred.

### Versions

* Last commit id on master:
* Operating System (lsb_release -a):

### Logs

Any logs (if any) generated in

#### SW360 logs

Logs generated under /var/log/sw360/sw360.log

#### Tomcat logs

Logs generated under /var/log/tomcat/error.log

### PR DESCRIPTION
**Pull request by keerthi-bl**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: 
Closes #2256 

URL : 
http://localhost:8080/resource/api/reports?withlinkedreleases=false&module=projects&projectId=bf1570a235414de486eb1043d4b0c0a8

Response : 
After send and download report should get downloaded.

> Have you implemented any additional tests?
No

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by keerthi-bl** [comment]
#2249 : Dependent PR.. Please review this after that PR got merged.

**Comment by nikkuma7** [comment]
Test successful.

image
image
Kindly pass the status message with response.

@keerthi-bl , As per our discussion once it deploys in stage please let know.

**Comment by nikkuma7** [comment]
image

**Comment by nikkuma7** [comment]
@keerthi-bl , If i pass a wrong project-id but response is getting 200, ideally status should come 404.

image

Im getting mail when i pass correct id, And when i open the link getting below error message.

image

NOTE: Testing in stage env.

**Comment by keerthi-bl** [comment]
> @keerthi-bl , If i pass a wrong project-id but response is getting 200, ideally status should come 404.
> 
> image
> 
> Im getting mail when i pass correct id, And when i open the link getting below error message.
> 
> image
> 
> NOTE: Testing in stage env.

All the issues addressed, please test with Bearer token in stage.

**Comment by nikkuma7** [comment]
@keerthi-bl, its working fine in stage, I get mail and download the xls sheet as well.
Note: Getting mail only for projects and components.

File: libraries/datahandler/src/main/resources/sw360.properties [hunk 1]
```diff
@@ -53,3 +53,6 @@
 ## This property is used to enable the dependency management function.
 ## To enable the function, you should also run the migration script (scripts/migrations/060_migrate_project_dependency_network.py).
 #enable.flexible.project.release.relationship=true
+
+## This property is used to enable or disable report should receive in mail or direct download.
+send.project.spreadsheet.export.to.mail.enabled=true
```
Comment (rudra-superrr) [inline libraries/datahandler/src/main/resources/sw360.properties:58]: why adding this field again as this is already added in backend's sw360.properties file?

File: rest/resource-server/src/docs/asciidoc/projects.adoc [hunk 2]
```diff
@@ -848,28 +848,11 @@ include::{snippets}/should_document_create_summary_administration/http-response.
...
-include::{snippets}/should_document_get_project_report_without_mail_req/curl-request.adoc[]
-
-===== Example response
-include::{snippets}/should_document_get_project_report_without_mail_req/http-response.adoc[]
-
-[[resources-projects-download-report-mail-request]]
-==== Downloading project report with mail request
-
-A `GET` request help to download the projects report with mail request true.
+A `GET` request help to download the projects report true.
```
Comment (rudra-superrr) [inline rest/resource-server/src/docs/asciidoc/projects.adoc:851]: please change it to, A `GET` request help to download the project report.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportController.java [hunk 3]
```diff
@@ -93,17 +94,19 @@ public void getProjectReport(
...
-                        break;
+                case PROJECTS:
+                    getProjectReports(withLinkedReleases, SW360Constants.MAIL_REQUEST_FOR_PROJECT_REPORT, response,
+                            request, sw360User, module, projectId);
+                    break;
+                case COMPONENTS:
+                    getComponentsReports(withLinkedReleases, SW360Constants.MAIL_REQUEST_FOR_COMPONENT_REPORT, response,
+                            request, sw360User, module);
+                    break;
+                case LICENSES:
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportController.java:105]: Do we need this case right now? as this PR is valid for projects and components only.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportService.java [hunk 4]
```diff
@@ -33,23 +33,43 @@ public class SW360ReportService {
...
 
     public ByteBuffer getProjectBuffer(User user, boolean extendedByReleases, String projectId) throws TException {
+        if (projectId != null && validateProject(projectId, user)) {
+            throw new TException("No project record found for the project Id : " + projectId);
+        }
         return projectclient.getReportDataStream(user, extendedByReleases, projectId);
     }
 
+    private boolean validateProject(String projectId, User user) throws TException {
+        boolean res = true;
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportService.java:43]: give a meaningful name.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportController.java (hunk 3 above)
Comment (keerthi-bl) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportController.java:105]: It has no changes with previous commit. It is fine as it is.

**Review by GMishx — APPROVED** [review]
Changes looks good.

File: libraries/datahandler/src/main/resources/sw360.properties (hunk 1 above)
Comment (GMishx) [reply libraries/datahandler/src/main/resources/sw360.properties]: @rudra-superrr @keerthi-bl , this field will be required when the rest server is separated?

Comment (keerthi-bl) [reply libraries/datahandler/src/main/resources/sw360.properties]: No @GMishx , its not required. It is added in different file path, there its enough.

Comment (GMishx) [reply libraries/datahandler/src/main/resources/sw360.properties]: In that case it can be deleted from here

Comment (keerthi-bl) [reply libraries/datahandler/src/main/resources/sw360.properties]: Deleted and resolved already!

Comment (GMishx) [reply libraries/datahandler/src/main/resources/sw360.properties]: Umm... I can still see the changes in PR

image

Maybe you forgot to add the changes to commit?

Comment (GMishx) [reply libraries/datahandler/src/main/resources/sw360.properties]: Oh, the properties are expected there. The comment is on libraries.

Sorry for the confusion.

**Pull request merged**